packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/athena/start-query-execution.ts

```diff
@@ -66,10 +76,21 @@ export class AthenaStartQueryExecution extends sfn.TaskStateBase {
     this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
 
     validatePatternSupported(this.integrationPattern, AthenaStartQueryExecution.SUPPORTED_INTEGRATION_PATTERNS);
+    this.validateExecutionParameters(props.executionParameters);
 
     this.taskPolicies = this.createPolicyStatements();
   }
 
+  private validateExecutionParameters(executionParameters?: string[]) {
+    if (executionParameters == null) return;
+    if (executionParameters.length == 0) {
+      throw new Error('Query executionParameters must not be an empty list');
+    }
+    if (executionParameters.length > 1024) {
+      throw new Error('Too many executionParameters for query, maximum 1024 supported');
+    }
```

Comment:
```suggestion
    const invalidExecutionParameters = executionParameters.some(p => p.length < 1 || p.length > 1024);
    if (invalidExecutionParameters) {
        throw new Error('\'executionParameters\' items\'s length must be between 1 and 1024 characters');
    }
```
> Length Constraints: Minimum length of 1. Maximum length of 1024.

This refers to the length of the single strings in the array.

Reply:
Good catch, didn't understand that from the documentation.